#164146041 Create REST endpoint for User Login (POST)

MicahBala opened the pull request:
### What does this PR do?
Send a POST request with username and password to login

### Description of tasks to be completed
have the endpoint below working
POST /auth/login

### How should this be manually tested?
Clone the repo, cd into /app/api and run `npm run start ` to start server locally
then use postman to send data below in json format or x-www-fomr-urlencoded:
```
{
   "userName":  "samjames"
   "password": "samjames"
}

```

### Any background context you want to provide?
This is part of a meal booking app

### What are the relevant pivotal tracker stories?
#164146041

MicahBala committed:
[Feature #164145932] Added endpoint to create new user

MicahBala committed:
[Feature #164145932] Write test for signup enpoint

MicahBala committed:
[Feature #164146041] Added endpoint for user signin

MicahBala committed:
[Feature #164146041] Wrote test for signin endpoint

MicahBala committed:
[Feature #164146041] Corrected code on meals.controller that generates error

Billmike commented on app/api/mealsController/users.controller.js:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    const userToAdd = {
+      id: usersDb.length + 1,
```
Again, refrain from using the length of the array for ID. Take a look at [shortId](https://www.npmjs.com/package/shortid)

Billmike commented on app/api/test/mytests.js:
```diff
@@ -85,8 +82,33 @@ describe("Test API endpoints", () => {
+          userName: "micahbala",
+          password: "micahbala",
+          isAdmin: false
+        })
```
You want to add robust assertions to your test cases. Your expect method should assert that the object created here has all the necessary fields (firstName, lastName, etc) in them

Billmike commented on app/api/test/mytests.js:
```diff
@@ -85,8 +82,33 @@ describe("Test API endpoints", () => {
+          userName: "micahbala",
+          password: "micahbala"
+        })
+        .expect(200, done);
```
Same feedback as above

MicahBala committed:
[Feature #164146041] Implemented reviewer Suggestions

Used shortId to generate ids for storage in model
Edit test file to assert that objects hass all necessary fields
